src/main/java/com/wavefront/datastructures/Distribution.java

```diff
@@ -36,48 +35,7 @@ public Distribution(int startValue, int endValue, double percentage) {
     this.startValue = startValue;
     this.endValue = endValue;
     this.percentage = percentage;
-  }
-
-  /**
-   * By use percentages, generating percent ranges.
-   */
-  protected static List<Double> generatePercentRanges(@Nonnull List<Double> percentages) {
-    List<Double> percentRanges = new LinkedList<>();
-    sumOfPercentages = 0;
-    for (int i = 0; i < percentages.size(); i++) {
-      sumOfPercentages += percentages.get(i);
-      percentRanges.add(sumOfPercentages);
-    }
-    return percentRanges;
-  }
-
-  /**
-   * Sum of all percentages.
-   */
-  protected static double sumOfPercentRanges() {
-    return sumOfPercentages;
-  }
-
-  /**
-   * Used binary search algorithm to find index.
-   */
-  protected static int getIndex(@Nonnull List<Double> percentRanges, double randomPercent) {
-    int firstIndex = 0;
-    int lastIndex = percentRanges.size() - 1;
-    int middleIndex = 0;
-
-    while (firstIndex <= lastIndex) {
-      middleIndex = firstIndex + (lastIndex - firstIndex) / 2;
-      if (percentRanges.get(middleIndex) == randomPercent) {
-        return middleIndex;
-      }
-      if (percentRanges.get(middleIndex) < randomPercent) {
-        firstIndex = middleIndex + 1;
-      } else {
-        lastIndex = middleIndex - 1;
-      }
-    }
-    return middleIndex;
+    this.portion = 0;
```

Comment:
if this field is always 0 at object instantiation moment, you can assign it on the declaration line.
public double portion = 0;